harvester_e2e_tests/integration/test_images.py

```diff
@@ -0,0 +1,212 @@
+    """
+    Test create raw and iso type image from url
+
+    Steps:
+    1. Open image page and select default URL
+    2. Input qcow2 image file download URL, wait for download complete
+    3. Check the qcow2 image exists
+    4. Input iso image file download URL, wait for download complete
+    5. Check the iso image exists
+    """
+
+    test_images.create_image_url(api_client, 'opensuse', qcow2_image_url, 600)
+    test_images.create_image_url(api_client, 'k3os', iso_image_url, 600)
+
+
+@pytest.mark.images_p1
+@pytest.mark.dependency(name="create_image_file")
+def test_upload_image_file(api_client):
+    """
+    Test create raw and iso type image from file
+
+    Steps:
+    1. Open image page and select default URL
+    2. Upload an qcow2 file type image from file
+    3. Check the the uploaded file exists
+    """
+
+    testimages = test_images.TestImages()
+
+    testimages.test_upload_file(api_client, 'test-file-upload', 'test-upload.qcow2', '10M')
+
+    testimages.test_get(api_client, 'test-file-upload')
+
+
+@pytest.mark.images_p1
+@pytest.mark.dependency(depends=["create_image_url", "create_image_file"])
+def test_delete_image_recreate(api_client, iso_image_url):
+    """
+    Test create raw and iso type image from file
+
+    Steps:
+    1. Check the image created by URL exists
+    2. Delete the newly created image
+    3. Create an iso file type image from URL
+    4. Check the iso image exists
+    5. Upload an qcow2 file type image
+    5. Delete the newly uploaded file
+    6. Upload a new qcow2 file type image
+    """
+    testimages = test_images.TestImages()
+
+    testimages.test_get(api_client, 'k3os')
+    testimages.test_delete(api_client, 'k3os', 15)
+
+    test_images.create_image_url(api_client, 'k3os', iso_image_url, 600)
+
+    testimages.test_get(api_client, 'test-file-upload')
+    testimages.test_delete(api_client, 'test-file-upload', 15)
+
+    testimages.test_upload_file(api_client, 'test-file-upload', 'test-upload.qcow2', '10M')
+
+
+@pytest.mark.images_p1
+def test_create_invalid_url(api_client):
+    """
+    Test create image from invalid url
+
+    Steps:
+    1. Create image with invalid URL. e.g. - https://test.img
+    2. Check the Image state show as Failed
+    """
+    test_images.test_create_with_invalid_url(api_client, 'test')
+
+
+@pytest.mark.images_p1
+def test_create_invalid_file(api_client):
+    """
+    Test create upload image from invalid file type
+
+    Steps:
+    1. Try to upload invalid image file to images page, something like dmg, or tar.gzps
+    2. Check should get an error
+    """
+
+    os.system('''
+        cat <<EOF >>test-invalid.tar.gzps
+        test
+        EOF
+    ''')
+    resp = api_client.images.create_by_file('test-invalid-file', 'test-invalid.tar.gzps')
+
+    assert 500 == resp.status_code, (
+        f"Failed to upload unsupported file type:{resp.status_code}, {resp.content}"
+    )
+
+    os.system('rm test-img.tar.gzps')
```

Comment:
this is not the right way to implement the test.
what if the test case failed? the `rm` command will never executed.

you should use NamedTemporaryFile[^1] instead.

[^1]: https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

Reply:
Add the file clean up in class level function `def teardown_class(self):` 

When using the `fake_image_file` fixture in tests 
  ```
  @pytest.mark.images_p1
      def test_fake_image(self, api_client, unique_name, fake_image_file):
  
          # testimages = test_images.TestImages()
          resp = api_client.images.create_by_file(unique_name, fake_image_file)
  
          assert 200 == resp.status_code, (
              f"Failed to upload fake image with error:{resp.status_code}, {resp.content}"
          )
  
  ``` 
Test execution encounter failure with error message
  ```
  Timeout waiting for the datasource file processing begin
  ```
  ![image](https://user-images.githubusercontent.com/29251855/209543135-3348774d-32f3-4ccb-b989-3ca6a88d7e5b.png)

We have related issue to track the status.  
https://github.com/harvester/harvester/issues/3259 

In order to create failed file type, thus I try using os command line as to achieve it.